### PR TITLE
Resolve only local intents in EndTransaction

### PR DESCRIPTION
**Pull request by kkaneda**

### CODE REVIEW

File: storage/replica_test.go [hunk 1]
```diff
@@ -1829,6 +1829,52 @@ func TestEndTransactionGC(t *testing.T) {
...
+	args.Timestamp = txn.Timestamp
+	args.Intents = []proto.Intent{{Key: key, EndKey: splitKey.Next()}}
+	if _, err := tc.rng.AddCmd(tc.rng.context(), &args); err != nil {
+		t.Fatal(err)
+	}
+
+	// Check if the intent in the other range has not yet been resolved.
+	gArgs := getArgs(splitKey, newRng.Desc().RangeID, tc.store.StoreID())
+	_, err := newRng.AddCmd(newRng.context(), &gArgs)
+	if err == nil {
```
Comment (tschottdorf) [inline storage/replica_test.go]: I think you can skip this. The check for `WriteIntentError` below should get it.

**Comment by tschottdorf** [comment]
Good catch. LGTM

**Comment by bdarnell** [comment]
LGTM

File: storage/replica_test.go (hunk 1 above)
Comment (kkaneda) [inline storage/replica_test.go]: Sure, removed.

**Pull request merged**